### DISCUSSION TITLE
Output lockfile version 3

### DISCUSSION THREAD

**Discussion by remcohaszing** [npm/feedback#517] [CLI] [answered]
I manage a project which uses workspaces. This means the minimum requirement for this project is npm 7. npm 7 outputs `package-lock.json` version 2. As far as I know, `package-lock.json` version 2 is an intermediate format with whose only goal is to provide backwards compatibility with npm 6 and `package-lock.json` version 7 by outputting both formats in the same file. The cost of this is that the entire dependency tree defined twice. This basically means the lockfile, which is already huge, now has twice the size that’s necessary for no benefit.

Can I make npm output `package-lock.json` version 3? I imagine something like `package-lock-version=3` in `.npmrc`. I couldn’t find any documentation on this subject.

For reference: [package-lock.json](https://docs.npmjs.com/cli/v7/configuring-npm/package-lock-json)

**Comment by MylesBorins** [comment, marked as answer]
@remcohaszing if you would like to propose a feature for the npm CLI the next step would be to open an RFC over at https://github.com/npm/rfcs
